Source/ExecutionEngine/ExecutionEngine.cs

```diff
@@ -1677,6 +1692,10 @@ private static void ProcessErrors(List<Counterexample> errors, VC.VCGen.Outcome
           {
             errorInfo.Out.WriteLine("Execution trace:");
             error.Print(4, errorInfo.Out, b => { errorInfo.AddAuxInfo(b.tok, b.Label, "Execution trace"); });
+            if (CommandLineOptions.Clo.PrintErrorModel >= 1 && error.Model != null)
+            {
+              error.Model.Write(ExecutionEngine.ModelWriter == null ? errorInfo.Out : ExecutionEngine.ModelWriter);
+            }
```

Comment:
1. Does the synchronization in `ExecutionEngine` ensure that there is no race on the initialization of `ExecutionEngine.ModelWriter`?
2. I was at first surprised that `ExecutionEngine.ModelWriter` is only initialized when writing to a file, until I saw that `errorInfo.Out` is used here when not writing to a file.
3. I saw that previously ModelWriter was flushed. Is this not necessary here?

Reply:
1. Good catch.  It does not.  I have changed the logic now so that ModelWriter is initialized at the beginning of InferAndVerify before all the parallelization starts.

3. The Flush is happening in OutputCollector.WriteMoreOutput.